docs/technologies.md

```diff
@@ -0,0 +1,423 @@
+    from bigflow.resources import find_or_create_setup_for_main_project_package, resolve, get_resource_absolute_path
+    options = PipelineOptions()
+
+    google_cloud_options = options.view_as(GoogleCloudOptions)
+    google_cloud_options.project = gcp_project_id
+    google_cloud_options.job_name = f'beam-wordcount-{uuid.uuid4()}'
+    google_cloud_options.staging_location = f"gs://{staging_location}"
+    google_cloud_options.temp_location = f"gs://{temp_location}"
+    google_cloud_options.region = region
+
+    options.view_as(WorkerOptions).machine_type = machine_type
+    options.view_as(WorkerOptions).max_num_workers = 2
+    options.view_as(WorkerOptions).autoscaling_algorithm = 'THROUGHPUT_BASED'
+    options.view_as(StandardOptions).runner = 'DataflowRunner'
+
+    options.view_as(SetupOptions).setup_file = resolve(find_or_create_setup_for_main_project_package(project_name, Path(__file__)))
+    options.view_as(SetupOptions).requirements_file = resolve(get_resource_absolute_path('requirements.txt', Path(__file__)))
+    return beam.Pipeline(options=options)
+```
+
+The `dataflow_pipeline` function create a Beam pipeline. The key is the following line:
+
+```python
+options.view_as(SetupOptions).setup_file = resolve(find_or_create_setup_for_main_project_package(project_name, Path(__file__)))
+```
+
+The line sets a path to the setup file, that will be used by Beam to create a package. The setup for a Beam
+process is generated on-fly when a Beam job is run. So it's not the `project_setup.py`. The generated setup looks like this:
+
+```python
+import setuptools
+
+setuptools.setup(
+        name='your_project_name',
+        version='0.1.0',
+        packages=setuptools.find_namespace_packages(include=["your_project_name.*"])
+)
+```
+
+The generated setup is minimalistic. If you want to provide requirements for your Beam process, you can do it through the
+`SetupOptions`. You can store the requirements for your processes in the [`resources`](./project_setup.py#project-structure) directory.
+
+```python
+options.view_as(SetupOptions).requirements_file = resolve(get_resource_absolute_path('requirements.txt', Path(__file__)))
+```
+
+## BigQuery
+
+### Overview
+
+BigFlow provides vast support for BigQuery. Example use cases:
+
+* Ad-hoc queries (fits well in a Jupyter notebook)
+* Convenient BigQuery client 
+* Creating data processing pipelines
+* Creating BigQuery sensors
+
+The project starter generates the workflow called `internationalports`. It's a purely BigQuery based workflow.
+
+The workflow fits the single `internationalports.workflow` module:
+
+```python
+from bigflow import Workflow
+from bigflow.bigquery import DatasetConfig, Dataset
+
+dataset_config = DatasetConfig(
+    env='dev',
+    project_id='your-project-id',
+    dataset_name='internationalports',
+    internal_tables=['ports', 'polish_ports'],
+    external_tables={})
+
+dataset: Dataset = dataset_config.create_dataset_manager()
```

Comment:
Why only this one variable has explicit type?

Reply:
To highlight that the created object is a Dataset object.

Reply:
variable name is `dataset` explicit type adds no additional info

Reply:
If someone reads the doc (and this piece of code) for the first time, "dataset" means nothing for such a person. Some people won't even read the text. I think that this annotation can only help, it won't harm anybody.

Reply:
w zwykłym kodzie nie deklarujemy typów dla zmiennych ale to snippet dokumentacyjny, więc jeśli to ma podnieść walory edukacyjne tego docsa to ja bym się nie upierał, żeby usuwać typ

Reply:
też sie nie upieram, że musi to zostać. No to do wywalenia :)